src/compressed_tensors/transforms/hadamard.py

```diff
@@ -46,13 +50,25 @@ def __init__(
         :param dtype: type to cast the rotation matrix to
 
         """
-        if not empty:
-            # TODO: this is deterministic; we should just serialize the size
-            transform = torch.Tensor(deterministic_hadamard_matrix(size=size))
+        self.matrix_registry = SingletonMatrixRegistry()
```

Comment:
Maybe it would be cleaner to implement this on the class level? That way instances don't have to carry around a reference to the matrix registry. For example,

```python3
class Hadamard(Transforms):
    registry: Dict[int, torch.Tensor] = {}

    def __new__(cls, size, empty, transform_name, *args, **kwargs):
        if empty:
            matrix = ...
        else:
            matrix = cls.registry.get(size, torch.Tensor(deterministic_hadamard_matrix(size=self.size)))
        return super().__new__(transform=matrix, transform_name=transform_name)
````